src/js/helpers.js

```diff
@@ -21,6 +24,16 @@ export function getActivePages(pages, data) {
   return pages.filter(page => isActivePage(page, data));
 }
 
+export function getActiveProperties(activePages) {
+  let allProperties = [];
```

Comment:
I tried implementing this the way you suggested, which I liked! However I was running into an issue with unit tests failing. If the test data isn't complete and doesn't include `schema` for every `page`, we get the following error: `Cannot read property 'properties' of undefined`. That's why I included a check for `if (page.schema)` exists before trying to return the keys.

I'm not sure what the best workaround for this is. I tried adding a check for `if (page.schema)` again in this `.map` function, but it doesn't work. I could fix it by filling out complete test data for every unit test, but that seems like a significant burden. It also may have future issues; should the code fail completely if someone forgets to write a complete config and there's a `page` without `schema`? Or if someone is just setting up the initial scaffolding? Not sure the best way to handle this situation.

Reply:
This is similar to the problem I had in #153 where the unit tests were too skeletal and not set up like a real form. I'd prefer to have a check up front that every `page` contains a `schema` so we don't need to include guards everywhere in the code for a case that definitely seems broken. Even if we just throw an error on the console it gives the developer an idea of what they did wrong, since we should be able to give a specific "Chapter X page Y property Z does not contain a required `schema`" message. 

At some point since we're pulling in a JSON Schema validator we should be able to run `formConfig` through that validator if we are brave enough to write a schema for it. 💫 

The only case I can think of where we might not have a schema is if we end up doing #150 and use a form page for the intro page. Even for that, if we need to we could eliminate the special case by injecting empty schema/properties into it.

Reply:
I like the idea of checking once up front for the presence of `schema`. Where do you think that check should happen? I might create a separate ticket for that.

Reply:
Good question. Is it enough to just check the `formConfig` and its related parts once at startup? If so maybe it could go at the top of `FormApp`. If we need/want to also check dynamic changes to `formConfig` and what's inside it, the first question I'd have is whether we are doing that directly or going through some sort of update function. If the latter then we could just validate the part that changed.

Reply:
Oh, I meant to say that it sounds like a good idea to make a ticket.